compiler/z/codegen/ConstantDataSnippet.cpp

```diff
@@ -406,71 +406,81 @@ uint8_t *
+//  DC  0x8020b380       ; Address Of Constant Pool
+//  DC  0x00000040       ; CP Index
+//  DC  0x00000000       ; Interface class
+//  DC  0x00000000       ; Method Index    (contains Method ptr if the target is direct)
+//  DC  0x00000000       ; Thunk address   (used if the target is direct dispatch)
+//  DC  0x00000000       ; Flags (resolved=1, unresolved = 0)
+//  DC  0x100211cfe90    ; lastCachedSlotField
 //
-//  0x00000100211cfe90     DC  0x100211cfea0    ; firstSlotField
-//  0x00000100211cfe98     DC  0x100211cfed0    ; lastSlotField
+//  DC  0x100211cfea0    ; firstSlotField
+//  DC  0x100211cfed0    ; lastSlotField
 //
 
-//  If we use L/C/BR sequence:
+//  If we use L/C/BR sequence with 4-slots (0-slot and 1-slot are also possible):
 
-//  0x00000100211cfea0     DC  0x00000000       ; Cached Class1
-//  0x00000100211cfea8     DC  0x00000000       ; Cached Method1
+//  DC  0x00000000       ; Cached Class1   (unused if the target is direct)
+//  DC  0x00000000       ; Cached Method1  (unused if the target is direct)
 //
-//  0x00000100211cfeb0     DC  0x00000000       ; Cached Class2
-//  0x00000100211cfeb8     DC  0x00000000       ; Cached Method2
+//  DC  0x00000000       ; Cached Class2   (unused if the target is direct)
+//  DC  0x00000000       ; Cached Method2  (unused if the target is direct)
 //
-//  0x00000100211cfec0     DC  0x00000000       ; Cached Class3
-//  0x00000100211cfec8     DC  0x00000000       ; Cached Method3
+//  DC  0x00000000       ; Cached Class3   (unused if the target is direct)
+//  DC  0x00000000       ; Cached Method3  (unused if the target is direct)
 //
-//  0x00000100211cfed0     DC  0x00000000       ; Cached Class4
-//  0x00000100211cfed8     DC  0x00000000       ; Cached Method4
+//  DC  0x00000000       ; Cached Class4   (unused if the target is direct)
+//  DC  0x00000000       ; Cached Method4  (unused if the target is direct)
 //
 //  and so on...
 
 // If we use CLFI / BRCL
-//  0x00000100211cfea0     DC  0x00000000       ; Patch slot of the first CLFI (Class 1)
-//  0x00000100211cfea0     DC  0x00000000       ; Patch slot of the second CLFI (Class 2)
-//  0x00000100211cfea0     DC  0x00000000       ; Patch slot of the third CLFI (Class 3)
+//  DC  0x00000000       ; Patch slot of the first CLFI (Class 1)
+//  DC  0x00000000       ; Patch slot of the second CLFI (Class 2)
+//  DC  0x00000000       ; Patch slot of the third CLFI (Class 3)
 // and so on ...
 
 //  31-Bit Layout.
-//  0x00000000211cfe50     DC  0x0000610ff8     ; Address Of Dispatch Glue
-//  0x00000000211cfe54     DC  0x00211cf58a     ; Call Site RA
-//  0x00000000211cfe58     DC  0x2020b380       ; Address Of Constant Pool
-//  0x00000000211cfe5c     DC  0x00000040       ; CP Index
-//  0x00000000211cfe60     DC  0x00000000       ; Interface class
-//  0x00000000211cfe64     DC  0x00000000       ; Method Index
+//  DC  0x0000610ff8     ; Address Of Dispatch Glue
+//  DC  0x00211cf58a     ; Call Site RA
```

Comment:
Can we remove the leading zeros so the address is actually 31-bit? Also is there anything different for 31-bit layout? If there are no different fields and just that all fields are 32-bit rather than 64-bit I say we omit this altogether to avoid confusion.

Reply:
There's no difference with 31-bit except the fields are all word sized. Will remove this.